### DISCUSSION TITLE
Problem with new BLE support for Raspberry Pico W and platform.io

### DISCUSSION THREAD

**Discussion by seldom-c** [tttapa/Control-Surface#1005] [Q&A]
Hello,

I'm having a problem compiling the latest CS git to test BLE support with my Raspberry Pico W. I haven't been able to understand the cause of the problem after several hours so i'm posting here in case you can reproduce. The problem happens trying to compile a test project with platform.io:

Compiling .pio/build/raspicow/lib8cf/ArduinoBLE/utility/HCICordioTransport.cpp.o
.pio/libdeps/raspicow/ArduinoBLE/src/utility/HCICordioTransport.cpp:25:10: fatal error: driver/CordioHCITransportDriver.h: No such file or directory 
25 | #include <driver/CordioHCITransportDriver.h> 

If I "fix" include paths then the problem arises during linking:

/Users/hku/.platformio/packages/toolchain-gccarmnoneeabi/bin/../lib/gcc/arm-none-eabi/9.2.1/../../../../arm-none-eabi/bin/ld: .pio/build/raspicow/lib8cf/libArduinoBLE.a(HCICordioTransport.cpp.o): in function `HCICordioTransportClass::end()':
HCICordioTransport.cpp:(.text._ZN23HCICordioTransportClass3endEv+0x20): undefined reference to `ble_cordio_get_hci_driver()'
/Users/hku/.platformio/packages/toolchain-gccarmnoneeabi/bin/../lib/gcc/arm-none-eabi/9.2.1/../../../../arm-none-eabi/bin/ld: HCICordioTransport.cpp:(.text._ZN23HCICordioTransportClass3endEv+0x24): undefined reference to `ble::CordioHCIDriver::terminate()'
...

I'm using the latest versions of platform.io, Arduino BLE and such. Also the bluetooth stack is enabled during compilation:

https://arduino-pico.readthedocs.io/en/latest/platformio.html#bluetooth-stack

Thanks in advance for any feedback you may have on this issue

**Comment by tttapa** [comment]
If you have the Pi Pico W board selected, PlatformIO should set the `ARDUINO_RASPBERRY_PI_PICO_W` macro.  
This causes Control Surface to select the correct Bluetooth stack. Otherwise, it falls back to the ArduinoBLE library.

It appears that the ArduinoBLE library is not supported by the arduino-pico core, which is why you're seeing the errors you posted, but this is not the main cause.

Please post the PlatformIO configuration you're using.

**Comment by tttapa** [comment]
I can't seem to reproduce the issue. Here's the `platformio.ini` file I'm using:

```ini
[env:pico]
platform = https://github.com/maxgerhardt/platform-raspberrypi.git
board = rpipicow
framework = arduino
board_build.core = earlephilhower
board_build.filesystem_size = 0m
build_flags = -DUSE_TINYUSB -DPIO_FRAMEWORK_ARDUINO_ENABLE_BLUETOOTH
lib_deps = https://github.com/tttapa/Control-Surface.git#main
lib_ignore = MIDIUSB
build_type = release
```

**Reply by seldom-c** [reply]
Oh right, it compiles using your config, thank you. I was actually trying to use the official arduino-pico core as you said. This is the last platform.io config I was testing, but I got into several other compilation problems with btstack:

[env:raspicow]
platform = raspberrypi
board = pico
framework = arduino
build_flags = -DPIO_FRAMEWORK_ARDUINO_ENABLE_BLUETOOTH -DARDUINO_RASPBERRY_PI_PICO_W -DENABLE_BLE
upload_port = /Volumes/RPI-RP2/
monitor_speed = 115200
lib_deps = 
  https://github.com/tttapa/Control-Surface.git#main
  https://github.com/bluekitchen/btstack.git
  https://github.com/Leont/crypt-rijndael.git
  kmackay/micro-ecc
lib_ignore = MIDIUSB

i'll be doing my test using the earlephilhower core for now, but let me know if you want me to do more testing to make it work with arduino-pico core

Thanks again

**Reply by tttapa** [reply]
The link you posted in your original message is part of the earlephilhower core documentation. That's also what I based my configuration on.

You seem to be using the https://github.com/platformio/platform-raspberrypi platform, which (as far as I can tell) does not support the RPi Pico W (only the RPi Pico and the Arduino Nano RP2040 Connect, see https://github.com/platformio/platform-raspberrypi/tree/develop/boards).
